### PR TITLE
improve platform conditional imports

### PR DESCRIPTION
**Pull request by tomerd**
motivation: more reliable conditional imports

changes: use canImport(module) instead of os(OS) to drive conditional imports

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test

File: Sources/Basics/HTTPClient.swift [hunk 1]
```diff
@@ -20,9 +20,7 @@ import TSCBasic
 
 #if canImport(Glibc)
 import Glibc
-#endif
-
-#if os(Windows)
+#elseif canImport(CRT)
```
Comment (abertelrud) [inline Sources/Basics/HTTPClient.swift:23]: Is there a risk of something called CRT in the search path on other platforms?

Comment (tomerd) [reply Sources/Basics/HTTPClient.swift:23]: yes, but I dont think its very realistic. we can always make it more complicated if/when we run into that

**Review by abertelrud — APPROVED** [review]

**Pull request merged**